File: VB/MyXtraGrid/Form1.vb
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace MyXtraGrid

    Public Partial Class Form1
        Inherits Form

        Private customers As BindingList(Of Customer)

        Public Sub New()
            InitializeComponent()
            FillGridDataSource()
        End Sub

        Private Sub FillGridDataSource()
            customers = New BindingList(Of Customer)()
            For i As Integer = 0 To 50 - 1
                Dim customer As Customer = New Customer()
                customer.Name = "Mike"
                If i Mod 2 = 0 Then customer.Name = "John"
                customer.Age = 70 - i
                customer.Weight = 50 + i
                customer.Height = 150 + i
                customers.Add(customer)
            Next

            myGridControl1.DataSource = customers
        End Sub
    End Class
End Namespace

File: CS/MyXtraGrid/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace MyXtraGrid {
    public partial class Form1 : Form {
        BindingList<Customer> customers;
        public Form1() {
            InitializeComponent();
            FillGridDataSource();
        }

        private void FillGridDataSource()
        {
            customers = new BindingList<Customer>();
            for (int i = 0; i < 50; i++)
            {
                Customer customer = new Customer();
                customer.Name = "Mike";
                if (i % 2 == 0)
                    customer.Name = "John";                
                customer.Age = 70 - i;
                customer.Weight = 50 + i;
                customer.Height = 150 + i;
                customers.Add(customer);
            }
            myGridControl1.DataSource = customers;
        }
    }
}